src/ui/modules/urlExport.ts

```diff
@@ -40,45 +41,63 @@ const addUrlExportRequestHeaders = (request: XMLHttpRequest, exportSettings: url
   }
 }
 
-const addUrlExportRequestEvents = (request: XMLHttpRequest) => {
-  // on error
-  request.onerror = (event) => {
-    // @ts-ignore
-    parent.postMessage({
+function requestErrorHandler() {
+  parent.postMessage(
+    {
       pluginMessage: {
         command: commands.closePlugin,
         payload: {
-          notification: '🚨 An error occurred while sending the tokens: check your settings & your server.'
-        }
-      } as PluginMessage
-    }, '*')
-  }
-  // show message on successful push
-  request.onload = (progressEvent: ProgressEvent) => {
-    // @ts-ignore
-    parent.postMessage({
+          notification:
+            "🚨 An error occurred while sending the tokens: check your settings & your server.",
+        },
+      } as PluginMessage,
+    },
+    "*"
+  );
+}
+
+function requestLoadedHandler(request: XMLHttpRequest) {
+  // @ts-ignore
+  parent.postMessage(
+    {
       pluginMessage: {
         command: commands.closePlugin,
         payload: {
-          notification: responeHandler(progressEvent.target as XMLHttpRequest)
-        }
-      } as PluginMessage
-    }, '*')
-  }
+          notification: responseHandler(request),
+        },
+      } as PluginMessage,
+    },
+    "*"
+  );
+}
+
+const addUrlExportRequestEvents = (request: XMLHttpRequest) => {
+  // on error
+  request.onerror = (_event) => {
+    requestErrorHandler();
+  };
+  // show message on successful push
+  request.onload = (progressEvent: ProgressEvent) => {
+    requestLoadedHandler(progressEvent.target as XMLHttpRequest);
+  };
 }
 
 const generateUrlExportRequestBody = (exportSettings: urlExportSettings, requestBody: urlExportRequestBody) => {
   let body
-  if (exportSettings.authType === config.key.authType.gitlabToken) {
-    body = new FormData()
-    body.append('token', exportSettings.accessToken)
-    body.append('ref', exportSettings.reference)
-    body.append('variables[FIGMA_EVENT_TYPE]', requestBody.event_type)
-    body.append('variables[FIGMA_CLIENT_PAYLOAD_TOKENS]', requestBody.client_payload.tokens)
-    body.append('variables[FIGMA_CLIENT_PAYLOAD_FILENAME]', requestBody.client_payload.filename)
-    body.append('variables[FIGMA_CLIENT_PAYLOAD_COMMIT_MESSAGE]', requestBody.client_payload.commitMessage)
-  } else {
-    body = JSON.stringify(requestBody, null, 2)
+  switch (exportSettings.authType) {
```

Comment:
Why do you feel a switch statement should be preferred here?

Reply:
I was thinking to handle the `gitlab_commit` authType here thats why i think switch is more suitable.

But I use `gitlabRepository.ts` to handle it.

Reply:
I've reverted it btw